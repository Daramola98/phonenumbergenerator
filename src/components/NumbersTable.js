import React from 'react';
import ReactTable from 'react-table';

import { formatDataForNumberTable } from '../utils';

const sortMethod = (a, b) => Number(a) - Number(b);

const NumbersTable = ({ numbersList }) => {
    const tableData = formatDataForNumberTable(numbersList);
    const columns = [
    {
      Header: 'NUMBERS',
      accessor: 'number',
      sortMethod
    }];

    return (
      <ReactTable
        data={tableData}
        columns={columns}
        minRows={1}
        sortable={true}
        defaultSorted={[
            {
              id: "number",
              desc: true
            }
          ]}
        className="-striped -highlight"
      />
    );
};

export default NumbersTable;