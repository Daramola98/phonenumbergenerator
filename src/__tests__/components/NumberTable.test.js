import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import NumbersTable from '../../components/NumbersTable';

Enzyme.configure({ adapter: new Adapter() });

describe('NumbersTable', () => {
  const numbersList = ["07011031609","07011031608","07011031649","07011031659","07011031689"];
  it('renders without crashing', () => {
    const wrapper = shallow(<NumbersTable numbersList={numbersList} />);
    expect(wrapper).toMatchSnapshot();
  });
});

