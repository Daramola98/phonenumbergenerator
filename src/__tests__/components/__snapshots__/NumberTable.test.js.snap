// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumbersTable renders without crashing 1`] = `ShallowWrapper {}`;
